core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -595,19 +596,22 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     // Precompute this because mPos will be updated several times in this function.
     final boolean isInCurrentBlock = pos / mBlockSize == mPos / mBlockSize;
 
-    // No need to cache if the stream hasn't read anything and the seek position is not within the
-    // block
-    if (mPos != 0 || mCurrentBlockInStream != null || isInCurrentBlock) {
+    // No need to cache if (1) the stream hasn't read anything and the seek position is not within
+    // the block, or (2) the in stream reads from the local worker
+    boolean firstReadAcrossBlock = mPos == 0 && mCurrentBlockInStream == null && !isInCurrentBlock;
```

Comment:
We can't use `mCurrentBlockInStream` as a boolean here, since we could be caching with the worker in case the source is from UFS

Reply:
Can you explain more on why `mCurrentBlockInStream ` cannot be used? The comment and the condition are equivalent, though the comment narrates in the negation.

Reply:
The variable name is confusing. It sounds like it should be true if we are seeking to a new block. However, why does it need the other predicates like the `mPos` one and the `mCurrentBlockInStream` one?

Reply:
Updated